File: frontend/src/models/ProviderPlaylist.ts
export interface ProviderPlaylist {
  id: string;
  providerId: string;
  title: string;
  description: string;
  thumbnailUrl: string;
  itemCount: number;
  privacyStatus: "private" | "public" | "unlisted";
  provider: number;
  channelId: string;
  channelTitle: string;
  createdAt: string;
}

File: backend/Application/Model/ProviderPlaylist.cs
using backend.Domain.Enum;

namespace backend.Application.Model;

public class ProviderPlaylist {
  public string Id { get; set; } = string.Empty;
  public string Title { get; set; } = string.Empty;
  public string Description { get; set; } = string.Empty;
  public string? ThumbnailUrl { get; set; }
  public string? ChannelId { get; set; }
  public string? ChannelTitle { get; set; }
  public DateTime CreatedAt { get; set; }
  public int ItemCount { get; set; }
  public string? PrivacyStatus { get; set; }
  public string ProviderId { get; set; } = string.Empty;
  public OAuthProvider Provider { get; set; }
}